### PR TITLE
[Feature]Support Alibaba DLF metastore for hive external table

### PR DESCRIPTION
**Pull request by mxdzs0612**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Data Lake Formation (DLF) is a key component of the cloud-native data lake framework which is widely used on Alibaba Cloud just like AWS Glue, details see https://www.alibabacloud.com/en/product/datalake-formation. This PR allows users to use DLF as metastore in Hive external table.

Usage:

```sql
CREATE EXTERNAL CATALOG dlf_hive_catalog
properties
(
    "type" = "hive",
    "hive.metastore.type" = "DLF"
);
```

Add a config file `hive-site.xml` to `{FE Home DIR}/conf` , with following configs:
```xml
<?xml version="1.0"?>
<configuration>
    <!--Set to use dlf client-->
    <property>
        <name>hive.metastore.type</name>
        <value>dlf</value>
    </property>
    <!--DLF endpoint, see https://www.alibabacloud.com/help/en/doc-detail/197608.html-->
    <property>
        <name>dlf.catalog.endpoint</name>
        <value>dlf-vpc.cn-xxx.aliyuncs.com</value>
    </property>
    <!--DLF region, see https://www.alibabacloud.com/help/en/doc-detail/197608.html-->
    <property>
        <name>dlf.catalog.region</name>
        <value>cn-beijing</value>
    </property>
    <!--Proxy mode of DLF-->
    <property>
        <name>dlf.catalog.proxyMode</name>
        <value>DLF_ONLY</value>
    </property>
    <!--Access Key mode of DLF-->
    <property>
        <name>dlf.catalog.akMode</name>
        <value>MANUAL</value>
    </property>
    <!--User id of the alibaba cloud account-->
    <property>
        <name>dlf.catalog.uid</name>
        <value>xxxxxx</value>
    </property>
    <!--Access Key ID of DLF, can be omitted if the cluster is created with the same Alibaba Cloud account of DLF-->
    <property>
        <name>dlf.catalog.accessKeyId</name>
        <value>xxxxxx</value>
    </property>
    <!--Access Key secret of DLF, can be omitted if the cluster is created with the same Alibaba Cloud account of DLF-->
    <property>
        <name>dlf.catalog.accessKeySecret</name>
        <value>xxxxxx</value>
    </property>
</configuration>
```

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 4 / 7 (57.14%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaStoreThriftClient.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaClient.java|4|6|66.67%|

File: fe/fe-core/pom.xml [hunk 1]
```diff
@@ -465,10 +465,30 @@ under the License.
 
         <!-- https://mvnrepository.com/artifact/com.facebook.presto.hive/hive-apache -->
         <dependency>
-            <groupId>com.facebook.presto.hive</groupId>
+            <groupId>io.trino.hive</groupId>
```
Comment (stephen-shelby) [inline fe/fe-core/pom.xml:468]: Please run the regression test.

File: fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaClient.java [hunk 2]
```diff
@@ -100,6 +101,7 @@ public class HiveMetaClient {
 
     public HiveMetaClient(String uris) throws DdlException {
         HiveConf conf = new HiveConf();
+        conf.addResource(new Path("file:///" + StarRocksFE.STARROCKS_HOME_DIR + "/conf/hive-site.xml"));
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaClient.java:104]: $FE/conf will be used as the classpath of fe. I think it is no necessary to add this?

File: fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaClient.java [hunk 3]
```diff
@@ -119,8 +121,13 @@ public class AutoCloseClient implements AutoCloseable {
         private final IMetaStoreClient hiveClient;
 
         private AutoCloseClient(HiveConf conf) throws MetaException {
-            hiveClient = RetryingMetaStoreClient.getProxy(conf, dummyHookLoader,
-                    HiveMetaStoreThriftClient.class.getName());
+            if ("dlf".equalsIgnoreCase(conf.get("hive.metastore"))) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaClient.java:124]: Where is the 'hive.metastore' configuration defined ? Could the name be modified? I think the 'hive.metastore' means too much

File: fe/pom.xml [hunk 4]
```diff
@@ -570,13 +570,21 @@ under the License.
...
                 <artifactId>hive-apache</artifactId>
-                <version>3.0.0-7</version>
+                <version>3.1.2-13</version>
             </dependency>
 
+            <!-- https://mvnrepository.com/artifact/com.aliyun.datalake/metastore-client-hive3 -->
+            <dependency>
+                <groupId>com.aliyun.datalake</groupId>
+                <artifactId>metastore-client-hive3</artifactId>
+                <version>0.2.14</version>
```
Comment (stephen-shelby) [inline fe/pom.xml:584]: plz define the ${metastore-client-hive3.version} in $STARROCKS/pom.xml

File: fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaStoreThriftClient.java [hunk 5]
```diff
@@ -723,6 +723,12 @@ public void createCatalog(Catalog catalog)
         throw new TException("method not implemented");
     }
 
+    @Override
+    public void alterCatalog(String s, Catalog catalog) throws NoSuchObjectException, InvalidObjectException,
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaStoreThriftClient.java:727]: Why add this interface

File: fe/fe-core/src/main/java/com/starrocks/external/hive/ProxyMetaStoreClient.java [hunk 6]
```diff
@@ -0,0 +1,2478 @@
...
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class ProxyMetaStoreClient implements IMetaStoreClient {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/external/hive/ProxyMetaStoreClient.java:153]: Is this only used by DLF?

File: fe/fe-core/src/main/java/com/starrocks/external/hive/ProxyMetaStoreClient.java [hunk 7]
```diff
@@ -0,0 +1,2478 @@
...
+    private static final Logger logger =
+            LoggerFactory.getLogger(com.aliyun.datalake.metastore.hive2.ProxyMetaStoreClient.class);
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/external/hive/ProxyMetaStoreClient.java:155]: pom use metastore-client-hive3 , but here we use hive2. is it ok?

File: fe/fe-core/src/main/java/com/starrocks/external/hive/ProxyMetaStoreClient.java [hunk 8]
```diff
@@ -0,0 +1,2478 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+// This file is based on code available under the Apache license here:
+//   https://github.com/aliyun/datalake-catalog-metastore-client/blob/master/metastore-client-hive/metastore-client-hive3/src/main/java/com/aliyun/datalake/metastore/hive2/ProxyMetaStoreClient.java
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/external/hive/ProxyMetaStoreClient.java:3]: @imay is license ok ?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 4 / 7 (57.14%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaStoreThriftClient.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaClient.java|4|6|66.67%|

**Comment by mxdzs0612** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 3 / 6 (50.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaStoreThriftClient.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaClient.java|3|5|60.00%|

**Review by stephen-shelby — COMMENTED** [review]
LGTM

**Review by stephen-shelby — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Review by caneGuy — APPROVED** [review]

**Pull request merged**